packages/is-shallow-equal/benchmark/index.js

```diff
@@ -13,55 +21,88 @@ const afterArraySame = beforeArray;
 const afterArrayEqual = [ 1, 2, 3, 4, 5, 6, 7 ];
 const afterArrayUnequal = [ 1, 2, 3, 4, 5, 'Unequal', 7 ];
 
-[
-	[
-		'@wordpress/is-shallow-equal (type specific)',
-		require( '..' ).isShallowEqualObjects,
-		require( '..' ).isShallowEqualArrays,
-	],
-	[ '@wordpress/is-shallow-equal', require( '..' ) ],
-	[ 'shallowequal', require( 'shallowequal' ) ],
-	[
-		'shallow-equal (type specific)',
-		require( 'shallow-equal/objects' ),
-		require( 'shallow-equal/arrays' ),
-	],
-	[ 'is-equal-shallow', require( 'is-equal-shallow' ) ],
-	[ 'shallow-equals', require( 'shallow-equals' ) ],
-	[ 'fbjs/lib/shallowEqual', require( 'fbjs/lib/shallowEqual' ) ],
-].forEach(
+Promise.all( [
+	lazyImport( 'shallowequal@^1.1.0' ),
+	lazyImport( 'shallow-equal@^1.2.1' ),
+	lazyImport( 'is-equal-shallow@^0.1.3' ),
+	lazyImport( 'shallow-equals@^1.0.0' ),
+	new Promise( async ( resolve ) => {
+		try {
+			await lazyImport( 'fbjs@^1.0.0' );
```

Comment:
For futere: it would be great to support natively packages that don’t have an entry point or custom path from the package.

Reply:
> For future: it would be great to support natively packages that don’t have an entry point or custom path from the package.

Yeah, the thought had come to mind, especially since the previous implementation was able to easily import from a nested path, whether it's something that could be supported in this implementation as well.

I'll give it a quick check to see if it might be straight-forward to add support. Otherwise, I can create an issue to follow up.

Reply:
> I'll give it a quick check to see if it might be straight-forward to add support. Otherwise, I can create an issue to follow up.

`npm-package-arg` doesn't seem to handle the path very well on its own. It might be something where we can either implement the directory parsing or the entire module / version spec parsing as something custom, but I'd suggest it be done separately.

Reply:
Thanks for opening a follow up issue 💯